cmd/thor/node/node_benchmark_test.go

```diff
@@ -38,6 +50,103 @@ func getCachedAccounts(b *testing.B) []genesis.DevAccount {
 	return cachedAccounts
 }
 
+func BenchmarkBlockProcess_RandomSigners_ManyClausesPerTx_RealDB(b *testing.B) {
+	// create state accounts
+	accounts := getCachedAccounts(b)
+
+	// randomly pick a signer for signing the transactions
+	randomSignerFunc := randomPickSignerFunc(accounts, createOneClausePerTx)
+
+	// create blocks
+	blocks := createBlocks(b, blockCount, accounts, randomSignerFunc)
+
+	// create test db
+	dbDir := "./bench-RandomSigners_ManyClauses/"
+	db, err := openTempMainDB(dbDir)
+	require.NoError(b, err)
+
+	// Ensure database cleanup after benchmark
+	defer func() {
+		err := os.RemoveAll(dbDir)
+		require.NoError(b, err, "failed to delete test database")
+	}()
+
+	// run the benchmark
+	benchmarkBlockProcess(b, db, accounts, blocks)
+}
+func BenchmarkBlockProcess_RandomSigners_OneClausePerTx_RealDB(b *testing.B) {
+	// create state accounts
+	accounts := getCachedAccounts(b)
+
+	// randomly pick a signer for signing the transactions
+	randomSignerFunc := randomPickSignerFunc(accounts, createManyClausesPerTx)
+
+	// create blocks
+	blocks := createBlocks(b, blockCount, accounts, randomSignerFunc)
+
+	// create test db
+	dbDir := "./bench-RandomSigners_OneClause/"
+	db, err := openTempMainDB(dbDir)
+	require.NoError(b, err)
+
+	// Ensure database cleanup after benchmark
+	defer func() {
+		err := os.RemoveAll(dbDir)
+		require.NoError(b, err, "failed to delete test database")
+	}()
+
+	// run the benchmark
+	benchmarkBlockProcess(b, db, accounts, blocks)
+}
+func BenchmarkBlockProcess_ManyClausesPerTx_RealDB(b *testing.B) {
+	// create state accounts
+	accounts := getCachedAccounts(b)
+
+	// Use one signer for signing the transactions
+	singleSignerFun := randomPickSignerFunc([]genesis.DevAccount{accounts[0]}, createManyClausesPerTx)
+
+	// create blocks
+	blocks := createBlocks(b, blockCount, accounts, singleSignerFun)
+
+	// create test db
+	dbDir := "./bench-ManyClauses/"
+	db, err := openTempMainDB(dbDir)
+	require.NoError(b, err)
+
+	// Ensure database cleanup after benchmark
+	defer func() {
+		err := os.RemoveAll(dbDir)
+		require.NoError(b, err, "failed to delete test database")
+	}()
+
+	// run the benchmark
+	benchmarkBlockProcess(b, db, accounts, blocks)
+}
+func BenchmarkBlockProcess_OneClausePerTx_RealDB(b *testing.B) {
+	// create state accounts
+	accounts := getCachedAccounts(b)
+
+	// Use one signer for signing the transactions
+	singleSignerFun := randomPickSignerFunc([]genesis.DevAccount{accounts[0]}, createOneClausePerTx)
+
+	// create blocks
+	blocks := createBlocks(b, blockCount, accounts, singleSignerFun)
+
+	// create test db
+	dbDir := "./bench-OneClause/"
```

Comment:
@otherview how about use the built in:

```golang
	dir := b.TempDir()
	defer os.Remove(dir)
```

or 

```golang
	dir := b.TempDir()
	b.Cleanup(func() { os.RemoveAll(dir) })
```

Reply:
Good idea ! Fixed !